barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_verifier.cpp

```diff
@@ -64,14 +154,98 @@ VerifierFoldingResult<typename VerifierInstances::Flavor> ProtoGalaxyVerifier_<
+    auto expected_next_target_sum =
+        perturbator_at_challenge * lagranges[0] + vanishing_polynomial_at_challenge * combiner_quotient_at_challenge;
+    auto next_target_sum = transcript->template receive_from_prover<FF>("next_target_sum");
+    bool verified = (expected_next_target_sum == next_target_sum);
+    auto expected_betas_star =
+        update_gate_challenges(perturbator_challenge, accumulator->folding_parameters.gate_challenges, deltas);
+    for (size_t idx = 0; idx < accumulator->log_instance_size; idx++) {
+        auto beta_star = transcript->template receive_from_prover<FF>("next_gate_challenge_" + std::to_string(idx));
+        verified = verified & (expected_betas_star[idx] == beta_star);
+    }
+
+    // Compute ϕ and verify against the data received from the prover
+    WitnessCommitments acc_witness_commitments;
+    auto witness_labels = commitment_labels.get_witness();
+    size_t comm_idx = 0;
+    for (auto& expected_comm : acc_witness_commitments.get_all()) {
+        expected_comm = Commitment::infinity();
+        size_t inst = 0;
+        for (auto& instance : instances) {
+            expected_comm = expected_comm + instance->witness_commitments.get_all()[comm_idx] * lagranges[inst];
+            inst++;
+        }
+        auto comm = transcript->template receive_from_prover<Commitment>("next_" + witness_labels[comm_idx]);
+        verified = verified & (comm == expected_comm);
+        comm_idx++;
+    }
+
+    std::vector<FF> folded_public_inputs(instances[0]->public_inputs.size(), 0);
+    size_t el_idx = 0;
+    for (auto& expected_el : folded_public_inputs) {
+        size_t inst = 0;
+        for (auto& instance : instances) {
+            expected_el += instance->public_inputs[el_idx] * lagranges[inst];
+            inst++;
+        }
+        auto el = transcript->template receive_from_prover<FF>("next_public_input" + std::to_string(el_idx));
+        verified = verified & (el == expected_el);
+        el_idx++;
+    }
+
+    auto expected_alpha = FF(0);
+    auto expected_parameters = proof_system::RelationParameters<FF>{};
+    for (size_t inst_idx = 0; inst_idx < VerifierInstances::NUM; inst_idx++) {
+        auto instance = instances[inst_idx];
+        expected_alpha += instance->alpha * lagranges[inst_idx];
+        expected_parameters.eta += instance->relation_parameters.eta * lagranges[inst_idx];
+        expected_parameters.beta += instance->relation_parameters.beta * lagranges[inst_idx];
+        expected_parameters.gamma += instance->relation_parameters.gamma * lagranges[inst_idx];
+        expected_parameters.public_input_delta +=
+            instance->relation_parameters.public_input_delta * lagranges[inst_idx];
+        expected_parameters.lookup_grand_product_delta +=
+            instance->relation_parameters.lookup_grand_product_delta * lagranges[inst_idx];
+    }
+
+    auto next_alpha = transcript->template receive_from_prover<FF>("next_alpha");
```

Comment:
Not necessary now, but like we do with the sumcheck evaluations we should put the relation parameters in e transcript as a single transcript entry.

Reply:
oh, that's a good idea, will add an issue